docs/adrs/flannel-options.md

```diff
@@ -0,0 +1,104 @@
+`--public-ipv6 (string)`: hardcoded to what is passed as k3s agent flag `--external-ip` when it is of IPv6 nature if k3s server flag `--flannel-external-ip` is true
+
+** Does not provide much value as flannel in k3s can configure the interface per node
+
+Regarding subnet&backend configuration:
+
+`Network` and `IPv6Network` are set by reading the assigned podCIDR for the Kubernetes node.
+`EnableIPv6` is set based on what the user passed as k3s server `--cluster-cidr`.
+`Backend` uses vxlan as default, although this can be changed using k3s server flag `--flannel-backend`
+
+Something important to note is that k3s allows the user to override the whole subnet&backend configuration, by using the k3s agent flag `--flannel-conf` and even the flannel CNI configuration file, by using the k3s agent flag `--flanel-cni-conf`. Overriding the whole subnet&backend configuration is useful if user wants to add specific backend configurations, e.g. VNI for vxlan. Users can currently do this by using `<=option1=val1,option2=val2>` when selecting the backend but it seldom used and complicates things, thus it was deprecated in favor of `--flannel-cni-conf`.
+
+To wrap up the context, k3s includes 3 agent flannel options:
+
+* flannel-iface (string): Overrides the default flannel interface. This interface is used to forward encapsulated traffic in inter-node communication. It matches the flannel flag `--iface`
+* flannel-conf (string): Path that points to a file containing the flannel subnet&backend config
+* flannel-cni-conf (string): Path that point to a flannel CNI config file. This config sets CNI specific configuration for that node
+
+And 3 server options:
+
+* flannel-backend (string): Sets the encapsulation technology that will allow inter-node traffic to work. It matches part of the Backend parameter in the subnet&backend configuration
+* flannel-ipv6-masq (bool): Enables masquerading traffic for IPv6. Note that IPv4 traffic is always masqueraded. It matches the ipv6 part of the `--ip-masq` flag
+* flannel-external-ip (bool): Enables using node external IP addresses for flannel traffic. It sort of matches on `--public-ip` and `--public-ipv6` flag. In this case, k3s selects the external-ips for those flags 
+
+
+### Design suggestion
+
+It was suggested to include a string slice flag that would consolidate all flannel server flags into just one. Benefits:
+* Simplify user's life by having less flags
+* Use it for potential new flannel server flags
+
+Downsides:
+* Users are used to those flags, specially `flannel-backend`, therefore it might be seen as a negative changei. That's why we should carefully follow the deprecation policy.
+
+## Proposal
+
+Starting in v1.26, introduce a new `flannel-opts` flag that includes flannel server options. The redundant flags are deprecated and removed in a few releases.
```

Comment:
OK, so just to be clear we're only talking about making changes to the **server** flags - the agent flags that allow users to pass custom flannel and cni config files would be retained?

Reply:
I see your point. I think I'm going to close this PR because it does not bring much user benefit. If in the future, there are more flannel flags coming (unlikely), we can reconsider this. But I think we are complicating things for just two flags, so the benefit is small